src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -636,6 +638,25 @@ public void unregisterVerticleFactory(VerticleFactory factory) {
   }
 
   @Override
+  public Vertx addSchedulerInterceptor(BiFunction<Context, Runnable, Runnable> interceptor) {
+    schedulerInteceptors.add(interceptor);
+    return this;
+  }
+
+  @Override
+  public Vertx removeSchedulerInterceptor(BiFunction<Context, Runnable, Runnable> interceptor) {
+    schedulerInteceptors.remove(interceptor);
+    return this;
+  }
+
+  @Override
+  public Runnable interceptScheduledWork(final ContextImpl context, final Runnable task) {
+    return schedulerInteceptors.stream()
+        .reduce((c, t) -> t, (a, b) -> (c, t) -> b.apply(c, a.apply(c, t)))
```

Comment:
Just an FYI. Since the set of interceptors is likely to be fixed after startup, I had wanted to cache the composition of the interceptors as an optimization (since, now, I'm rebuilding on each task schedule which feels pretty stupid). I didn't to avoid the cache maintenance logic and just minimize the number of changes while evaluating the concept.